lib/Doctrine/ORM/Internal/Hydration/SimpleObjectHydrator.php

```diff
@@ -99,13 +99,21 @@ protected function hydrateRowData(array $sqlResult, array &$result)
             }
 
             $discrMap = $this->class->discriminatorMap;
+            $discrType = $discriminatorValue = $this->class->discriminatorColumn['type'];
 
-            if ( ! isset($discrMap[$sqlResult[$discrColumnName]])) {
+            if (Type::hasType($discrType)) {
```

Comment:
This should always be `true`: we actually want an exception to be thrown if it is not.